src/Hazelcast.Net.Tests/Serialization/Compact/CSharpRecordTests.cs

```diff
@@ -0,0 +1,351 @@
+    {
+        var type = typeof(RecordClassWithPositionalProperty);
+        var rec = (RecordClassWithPositionalProperty) FormatterServices.GetUninitializedObject(type);
+
+        var properties = type.GetProperties();
+        Assert.That(properties.Length, Is.EqualTo(1));
+        var valueProperty = properties[0];
+        Assert.That(valueProperty.CanRead);
+
+        Assert.That(valueProperty.CanWrite);
+        valueProperty.SetValue(rec, "changed");
+        Assert.That(rec.Value, Is.EqualTo("changed"));
+
+        // it still is possible to implement other constructors but they all
+        // will have to invoke the automatic constructor (with positional
+        // properties) *and* that one *cannot* be modified - so we are safe
+
+        var fields = type.GetFields(BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic);
+        foreach (var field in fields) Console.WriteLine($">>> {field.Name}");
+        var backingFieldRegexp = new Regex("\\<(.+)\\>k__BackingField", RegexOptions.Compiled);
+        var x = fields.Select(x =>
+        {
+            var m = backingFieldRegexp.Match(x.Name);
+            return m.Success ? (x, m.Groups[1].Value) : (x, null);
+        }).ToArray();
+
+        Assert.That(x.Length, Is.EqualTo(1));
+        Assert.That(x[0].Item2, Is.EqualTo("Value"));
+
+        // BUT!
+
+        // a typical record (if it has not been altered) *does* have a init setter
+        // for all positional properties - so we can say that compact does *not*
+        // support properties that are get-only, full stop.
+    }
+
+    [Test]
+    public void GetUninitializedObjectDoesNotInitialize()
+    {
+        var type = typeof(ClassWithNoParameterLessConstructor);
+        var rec = (ClassWithNoParameterLessConstructor) FormatterServices.GetUninitializedObject(type);
+
+        // problem is, *nothing* is initialized
+        // it's not only the constructor that does not run
+
+        Assert.That(rec.Value, Is.Null);
+        Assert.That(rec.Value2, Is.Null);
+        Assert.That(rec._value3, Is.Null);
+    }
+
+    [Test]
+    public void GetProperConstructor()
+    {
+        var type = typeof (RecordClassWithPositionalProperty);
+
+        var ctors = type.GetConstructors();
+        var properties = type.GetProperties();
+
+        var serializableProperties = properties
+            .Where(x => x.CanRead && x.CanWrite)
+            .ToArray();
+
+        var serializablePropertyNames = new HashSet<string>(serializableProperties.Select(x => x.Name));
+
+        bool IsMatch(ConstructorInfo ctor)
+        {
+            foreach (var parameter in ctor.GetParameters())
+            {
+                if (!serializablePropertyNames.Contains(parameter.Name))
+                    return false;
+            }
+
+            return true;
+        }
+
+        var ctor = ctors.Where(IsMatch).OrderBy(x => x.GetParameters().Length).FirstOrDefault();
```

Comment:
I was referring line 40 in this file. I just confused with `OrderBy` here. Why do you get ctor with minimum number of matched parameters with `serializablePropertyNames`. Shouldn't be the otherwise? Ctor that matches with the most number of properties with in `serializablePropertyNames`. I mean why `FirstOrDefault()` bu `LastOrDefault()` .

Reply:
Ah... that's what we do in actual code byt here it's just a test where we want to ensure we've found the only one ctor that exist... so it does not make any difference...